### DISCUSSION TITLE
Unable to use MSI installer on Windows

### DISCUSSION THREAD

**Discussion by PratimGhosh86** [JanDeDobbeleer/oh-my-posh#5974] [Troubleshoot]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

```
Edition	Windows 11 Pro
Version	23H2
Installed on	‎11-‎May-‎22
OS build	22635.4515
Experience	Windows Feature Experience Pack 1000.22700.1061.0
```

Following error message is displayed when running the latest MSI installer

![image](https://github.com/user-attachments/assets/0baa8f3a-7744-436a-9e7b-e750a1e60be1)

https://github.com/JanDeDobbeleer/oh-my-posh/releases/tag/v24.8.0 exe installer works fine

### Theme

probua.minimal

### What OS are you seeing the problem on?

Windows

### Which shell are you using?

powershell

### Log output

```Shell
None. Unable to run installer
```

**Comment by JanDeDobbeleer** [comment]
@PratimGhosh86 I can't reproduce this on the latest installer on my Windows machine (nor when doing a clean install). Are you sure you're using the correct architecture?

**Reply by PratimGhosh86** [reply]
I believe so:

![image](https://github.com/user-attachments/assets/d56d3e64-9676-4a73-8759-cf2e8a1b1dc6)

I also tried via winget and that results in the following problem:

![image](https://github.com/user-attachments/assets/e7a94c4d-57b6-4b1a-99f2-db6aa931c540)

**Reply by JanDeDobbeleer** [reply]
Weird, because winget validates the installers. So theoretically, if I broke something, that would result in an error on their end.

Can you try uninstalling, then installing latest from the releases here?

**Reply by JanDeDobbeleer** [reply]
@PratimGhosh86 additionally, try enabling the logging by invoking from the CLI using the `/l*` switch

**Reply by PratimGhosh86** [reply]
I already did try a fresh install, but that didn't work either. 

And here's the log output if that contains some clue: [install-x64.msi.log.txt](https://github.com/user-attachments/files/17970206/install-x64.msi.log.txt)

**Reply by JanDeDobbeleer** [reply]
@PratimGhosh86 I checked our users on discord and nobody else can reproduce this either. Everyone can install the latest version without issues which is why I'm pretty sure the installer itself is OK, but something on your machine is influencing this negatively. The logs don't show anything specific, can you try running it full verbose?

**Reply by PratimGhosh86** [reply]
I did attach the log generated with `/l*`. The following entry seems strange:

```
DEBUG: Error 2330:  Error getting file attributes: C:\Users\Pratim\AppData\Local\Programs\oh-my-posh\bin. GetLastError: 448
The installer has encountered an unexpected error installing this package. This may indicate a problem with this package. The error code is 2330. The arguments are: 448, C:\Users\Pratim\AppData\Local\Programs\oh-my-posh\bin, 
MSI (s) (50:28) [08:40:48:466]: Product: Oh My Posh -- The installer has encountered an unexpected error installing this package. This may indicate a problem with this package. The error code is 2330. The arguments are: 448, C:\Users\Pratim\AppData\Local\Programs\oh-my-posh\bin, 
```

I also tried out some other MSI installers and those seem to be working fine

**Reply by JanDeDobbeleer** [reply]
@PratimGhosh86 from what I can read online it seems to be a permissions issue. The installer is unable to access the folder, can you try running it as admin?

**Reply by PratimGhosh86** [reply]
run MSI as admin? I did run it from an elevated command prompt but faced the same issue. The log file has the same error

**Reply by JanDeDobbeleer** [reply]
@PratimGhosh86 either way, it's a local issue as nobody's else can reproduce it. Try deleting that folder and install again. Something seems to have a lock or the permissions are off.

**Reply by PratimGhosh86** [reply]
Yeah seems so. Let me keep digging into this and see if anything works out

**Comment by PratimGhosh86** [comment]
Here is a quick update on this issue:

I'm still unable to use any of the MSI installers or install via winget

But, if I remove all the environment variables, and delete the installation directory, I am able to get it installed via Microsoft Store - the update doesn't work though. I need to remove the environment variables and the delete the installation folder and then re-install

I really have no idea why this is happening. I've tried multiple other MSI installers in the meantime which continue to work normally

**Reply by JanDeDobbeleer** [reply]
I also have no clue why this occurs, it's a very device specific issue as far as I can tell
